chore: Connection pooling issue in Maven may cause build failure

I ran across a flaky failure in the CI today, see here: https://github.com/INRIA/spoon/runs/1661992546

```
Error:  Failed to execute goal on project spoon-core: Could not resolve dependencies for project fr.inria.gforge.spoon:spoon-core:jar:8.4.0-SNAPSHOT: Could not transfer artifact com.martiansoftware:jsap:jar:2.1 from/to central (https://repo.maven.apache.org/maven2): Transfer failed for https://repo.maven.apache.org/maven2/com/martiansoftware/jsap/2.1/jsap-2.1.jar: Connection reset -> [Help 1]
Error:  
Error:  To see the full stack trace of the errors, re-run Maven with the -e switch.
Error:  Re-run Maven using the -X switch to enable full debug logging.
Error:  
Error:  For more information about the errors and possible solutions, please read the following articles:
Error:  [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/DependencyResolutionException
Error: Process completed with exit code 1.
```

It appears to be related to a connection pooling issue discussed in actions/virtual-environments#1499. There's a fix suggested there, in disabling Maven's HTTP pooling with `-Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false` in the GitHub Actions workflows. This however slows down the build by about 10-20% (I tried it). I've only run across this issue once, and so that build time hit seems like a high price to pay for it.

I wanted to make note of it anyway in case it happens again.

--- a/.github/workflows/tests.yml
+++ b/.github/workflows/tests.yml
@@ -38,11 +38,13 @@ jobs:
         run: mv chore/travis/logback.xml src/test/resources/
       - name: Build and test
         env:
-          MAVEN_OPTS: -Djava.src.version=${{ matrix.java }}
+          MAVEN_OPTS: -Djava.src.version=${{ matrix.java }} -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false
         run: mvn test
 
   coverage:
     runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false
     name: Test with coverage
     steps:
       - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4
@@ -59,6 +61,8 @@ jobs:
 
   extra:
     runs-on: ubuntu-latest
+    env:
+      MAVEN_OPTS: -Dhttp.keepAlive=false -Dmaven.wagon.http.pool=false
     name: Extra checks
     steps:
       - uses: actions/checkout@5a4ac9002d0be2fb38bd78e4b4dbde5606d7042f # v2.3.4

